base/linalg/symmetric.jl

```diff
@@ -364,8 +364,8 @@ function _inv(A::HermOrSym)
     end
     B
 end
-inv(A::Hermitian{T,S}) where {T<:BlasFloat,S<:StridedMatrix} = Hermitian{T,S}(_inv(A), A.uplo)
-inv(A::Symmetric{T,S}) where {T<:BlasFloat,S<:StridedMatrix} = Symmetric{T,S}(_inv(A), A.uplo)
+inv(A::Hermitian{<:Any,<:StridedMatrix}) = Hermitian(_inv(A), Symbol(A.uplo))
```

Comment:
Re my comment above regarding user define array types, should this be changed to:
```julia
inv(A::Hermitian{<:Any,S}) where {S<:StridedMatrix} = (Ai = _inv(A); Hermitian{eltype(Ai), S}(Ai, A.uplo)
```
such that we enforce a conversion back to the same array type? I noticed that if you don't implement `getindex` to return your user defined array you get back an `Array` from the backsolve.